vim-patch:8.2.{1506,1600,1624,1626,1751,2606,4679}

zeertzjq opened the pull request:
#### vim-patch:8.1.1949: cannot scroll a popup window to the very bottom

Problem:    Cannot scroll a popup window to the very bottom.
Solution:   Scroll to the bottom when the "firstline" property was set to -1.
            Allow resetting min/max width/height.
https://github.com/vim/vim/commit/8c6173c7d3431dd8bc2b6ffc076ef49512a7e175

Co-authored-by: Bram Moolenaar <Bram@vim.org>


#### vim-patch:8.2.1751: using 2 where bool is expected may throw an error

Problem:    Using 2 where bool is expected may throw an error.
Solution:   Make this backwards compatible.

https://github.com/vim/vim/commit/bade44e5cad1b08c85d4a8ba08d94a30458dddfb

In legacy Vim script get_bool functions do the same thing as get_number
functions, so just add aliases using #define.

N/A patches for version.c:

vim-patch:8.2.1506: Vim9: no error when using a number other than 0 or 1 as bool

Problem:    Vim9: no error when using a number other than 0 or 1 as bool.
Solution:   Check the number is 0 or 1.

https://github.com/vim/vim/commit/d70840ed68296c1144d743e6335003c81c558c24

Co-authored-by: Bram Moolenaar <Bram@vim.org>


#### vim-patch:8.2.1600: Vim9: cannot use "true" with deepcopy()

Problem:    Vim9: cannot use "true" with deepcopy().
Solution:   Use tv_get_bool_chk().

https://github.com/vim/vim/commit/44b4a246b62e0622550b963bcf3034dce3bcfc0c

Co-authored-by: Bram Moolenaar <Bram@vim.org>


#### vim-patch:8.2.1624: Vim9: cannot pass "true" to split(), str2nr() and strchars()

Problem:    Vim9: cannot pass "true" to split(), str2nr() and strchars().
Solution:   Use tv_get_bool_chk().

https://github.com/vim/vim/commit/3986b94b090ea258109630008611230a599999ab

Co-authored-by: Bram Moolenaar <Bram@vim.org>


#### vim-patch:8.2.1626: test for strchars() fails with different error number

Problem:    Test for strchars() fails with different error number.
Solution:   Adjust the error number.

https://github.com/vim/vim/commit/707be5f3524accb8b36e80bd2532e00b8246df55

Co-authored-by: Bram Moolenaar <Bram@vim.org>


#### vim-patch:8.2.2606: strchars() defaults to counting composing characters

Problem:    strchars() defaults to counting composing characters.
Solution:   Add strcharlen() which ignores composing characters.

https://github.com/vim/vim/commit/70ce8a1561c5396e4c4381f76a005cbb97646f80

Use docs from latest Vim instead.

Co-authored-by: Bram Moolenaar <Bram@vim.org>


#### vim-patch:8.2.4679: cannot have expandcmd() give an error message for mistakes

Problem:    Cannot have expandcmd() give an error message for mistakes.
Solution:   Add an optional argument to give errors. Fix memory leak when
            expanding files fails. (Yegappan Lakshmanan, closes vim/vim#10071)
https://github.com/vim/vim/commit/2b74b6805b5c8c4836b66df5d949f5ff6a77f8c7

Co-authored-by: Yegappan Lakshmanan <yegappan@yahoo.com>

zeertzjq committed:
vim-patch:8.2.4679: cannot have expandcmd() give an error message for mistakes

Problem:    Cannot have expandcmd() give an error message for mistakes.
Solution:   Add an optional argument to give errors. Fix memory leak when
            expanding files fails. (Yegappan Lakshmanan, closes vim/vim#10071)
https://github.com/vim/vim/commit/2b74b6805b5c8c4836b66df5d949f5ff6a77f8c7

Co-authored-by: Yegappan Lakshmanan <yegappan@yahoo.com>

zeertzjq committed:
vim-patch:8.1.1949: cannot scroll a popup window to the very bottom

Problem:    Cannot scroll a popup window to the very bottom.
Solution:   Scroll to the bottom when the "firstline" property was set to -1.
            (closes vim/vim#4577)  Allow resetting min/max width/height.
https://github.com/vim/vim/commit/8c6173c7d3431dd8bc2b6ffc076ef49512a7e175

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq commented:
Not sure if this is the best way to deal with the `get_bool` functions.

justinmk commented:
> In legacy Vim script get_bool functions do the same thing as get_number
> functions, so just add aliases using #define.

seems fine, or maybe `inline` function.

zeertzjq committed:
vim-patch:8.2.1751: using 2 where bool is expected may throw an error

Problem:    Using 2 where bool is expected may throw an error.
Solution:   Make this backwards compatible.

https://github.com/vim/vim/commit/bade44e5cad1b08c85d4a8ba08d94a30458dddfb

In legacy Vim script get_bool functions do the same thing as get_number
functions, so just add aliases using #define.

N/A patches for version.c:

vim-patch:8.2.1506: Vim9: no error when using a number other than 0 or 1 as bool

Problem:    Vim9: no error when using a number other than 0 or 1 as bool.
Solution:   Check the number is 0 or 1.

https://github.com/vim/vim/commit/d70840ed68296c1144d743e6335003c81c558c24

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq committed:
vim-patch:8.2.1624: Vim9: cannot pass "true" to split(), str2nr() and strchars()

Problem:    Vim9: cannot pass "true" to split(), str2nr() and strchars().
Solution:   Use tv_get_bool_chk(). (closes vim/vim#6884, closes vim/vim#6885, closes vim/vim#6886)

https://github.com/vim/vim/commit/3986b94b090ea258109630008611230a599999ab

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq committed:
vim-patch:8.2.1626: test for strchars() fails with different error number

Problem:    Test for strchars() fails with different error number.
Solution:   Adjust the error number.

https://github.com/vim/vim/commit/707be5f3524accb8b36e80bd2532e00b8246df55

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq committed:
vim-patch:8.2.2606: strchars() defaults to counting composing characters

Problem:    strchars() defaults to counting composing characters.
Solution:   Add strcharlen() which ignores composing characters.

https://github.com/vim/vim/commit/70ce8a1561c5396e4c4381f76a005cbb97646f80

Use docs from latest Vim instead.

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq committed:
vim-patch:8.2.1600: Vim9: cannot use "true" with deepcopy()

Problem:    Vim9: cannot use "true" with deepcopy().
Solution:   Use tv_get_bool_chk(). (closes vim/vim#6867)

https://github.com/vim/vim/commit/44b4a246b62e0622550b963bcf3034dce3bcfc0c

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq committed:
fix(eval): make error number of charidx() same as Vim
